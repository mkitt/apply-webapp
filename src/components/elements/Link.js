// @flow
import React, { type Node } from 'react'
import { Link } from 'react-router-dom'
import glamorous from 'glamorous'
import { isNotNil } from '../../lib/utils'
import { BASE_URL } from '../../constants/networking'

type Props = {
  children: Node,
  href?: string | null,
  to?: string | null,
}

const LinkComponent = ({ children, href, to, ...props }: Props) => {
  if (isNotNil(to)) {
    const url = (to || '').includes(BASE_URL) ? to : `${BASE_URL}${(to || '')}`
    return (
      <RouterLink to={url} {...props}>
        {children}
      </RouterLink>
    )
  }
  return (
    <ExternalLink href={href} target="_blank" rel="noopener noreferrer" {...props}>
      {children}
    </ExternalLink>
  )
}

LinkComponent.defaultProps = {
  href: null,
  to: null,
}

export default LinkComponent

// -------------------------------------

const RouterLink = glamorous(Link)()

const ExternalLink = glamorous.a()
