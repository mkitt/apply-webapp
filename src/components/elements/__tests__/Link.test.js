/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Link from '../Link'

describe('<Link />', () => {
  it('shows the correct snapshot tree for the <Link href> element', () => {
    const component = (
      <MemoryRouter>
        <Link href="https://www.turing.io">The External Link</Link>
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Link to> element', () => {
    const component = (
      <MemoryRouter>
        <Link to="/hello">The Router Link</Link>
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
