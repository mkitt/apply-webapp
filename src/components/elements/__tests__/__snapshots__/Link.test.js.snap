// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> shows the correct snapshot tree for the <Link href> element 1`] = `
<a
  className="css-nil"
  href="https://www.turing.io"
  rel="noopener noreferrer"
  target="_blank"
>
  The External Link
</a>
`;

exports[`<Link /> shows the correct snapshot tree for the <Link to> element 1`] = `
<a
  className="css-nil"
  href="/hello"
  onClick={[Function]}
>
  The Router Link
</a>
`;
